client/src/extension.ts

```diff
@@ -24,14 +26,14 @@ export function activate(context: ExtensionContext) {
     debug: {
       module: serverModule,
       transport: TransportKind.ipc,
-      options: debugOptions
-    }
+      options: debugOptions,
+    },
   };
 
   // Options to control the language client
   let clientOptions: LanguageClientOptions = {
     // Register the server for plain text documents
-    documentSelector: [{ scheme: 'file', language: 'avroavsc' }],
+    documentSelector: [{ language: "avroavsc" }],

```

Comment:
The real change here: remove "scheme: 'file'" so that the extension will be able to validate untitled: scheme documents which have had the language type set properly.